pkg/config/types_test.go

```diff
@@ -0,0 +1,170 @@
+						return multusConfig.CriSocketPath
+					}, Equal(containerdSocketPath)))
+			})
+		})
+
+		Context("overriding the configuration defaults", func() {
+			BeforeEach(func() {
+				Expect(
+					os.WriteFile(
+						configurationFilePath(configurationDir),
+						[]byte(genericConfigString(criSocketPath, multusSocketDir, cri.Crio)), allowAllPermissions),
+				).To(Succeed())
+			})
+			It("features the expected CRI socket path and multus socket directory", func() {
+				Expect(
+					LoadConfig(configurationFilePath(configurationDir)),
+				).To(
+					Equal(crioConfig(criSocketPath, multusSocketDir)))
+			})
+		})
+	})
+
+	It("fails when the config file is not present", func() {
+		const aPath = "non-existent-path"
+		_, err := LoadConfig(configurationFilePath(aPath))
+		Expect(err).To(MatchError(nonExistentPathError(aPath)))
+	})
+
+	It("fails when the config file does not feature valid json", func() {
+		Expect(
+			os.WriteFile(
+				configurationFilePath(configurationDir),
+				[]byte("invalid-json"), allowAllPermissions),
+		).To(Succeed())
+
+		_, err := LoadConfig(configurationFilePath(configurationDir))
+		Expect(err).To(MatchError(HavePrefix("failed to unmarshall the daemon configuration:")))
+	})
+
+	It("fails when the config file does not feature valid json", func() {
+		Expect(
+			os.WriteFile(
+				configurationFilePath(configurationDir),
+				[]byte("invalid-json"), allowAllPermissions),
+		).To(Succeed())
+
+		_, err := LoadConfig(configurationFilePath(configurationDir))
+		Expect(err).To(MatchError(HavePrefix("failed to unmarshall the daemon configuration:")))
+	})
+
+	It("fails when the config file features an invalid runtime", func() {
+		Expect(
+			os.WriteFile(
+				configurationFilePath(configurationDir),
+				[]byte(genericConfigString("", "", "pony")), allowAllPermissions),
+		).To(Succeed())
+
+		_, err := LoadConfig(configurationFilePath(configurationDir))
+		Expect(err).To(MatchError("invalid CRI type: pony. Allowed values are: containerd,crio"))
+	})
+})
+
+func nonExistentPathError(configDir string) string {
+	return fmt.Sprintf("failed to read the config file's contents: open %s/dummyconfig: no such file or directory", configDir)
+}
+
+func configContainerRuntime(multusConfig *Multus) cri.RuntimeType {
+	return multusConfig.CriType
+}
+
+func crioConfig(criSocketPath string, multusSocketDir string) *Multus {
+	return &Multus{
+		CriSocketPath: criSocketPath,
+		CriType:       cri.Crio,
+		SocketDir:     multusSocketDir,
+	}
+}
+
+func genericConfigString(criSocketPath string, multusSocketDir string, runtime cri.RuntimeType) string {
+	return fmt.Sprintf(`
+{
+    "criSocketPath": "%s",
+    "multusSocketDir": "%s",
+    "criType": "%s"
+}`, criSocketPath, multusSocketDir, runtime)
+}
+
+func configurationString(criSocketPath string, multusSocketDir string) string {
+	return fmt.Sprintf(`
+{
+    "criSocketPath": "%s",
+    "multusSocketDir": "%s"
+}`, criSocketPath, multusSocketDir)
+}
```

Comment:
Hard to understand the difference between the two functions from their names

Reply:
Done.